src/modules/course/course.service.ts

```diff
@@ -156,4 +164,74 @@ export class CourseService {
       throw new InternalServerErrorException(e);
     }
   }
+
+  // Create a Review
+  async addReview(
+    courseId: Schema.Types.ObjectId,
+    createReviewDto: CreateReviewDto,
+  ): Promise<any> {
+    try {
+      const course = await this.CourseModel.findById(courseId);
+      if (course) {
+        const newReview = new this.ReviewModel(createReviewDto);
+        await newReview.save();
+        course.reviews.push(newReview);
+        await course.save();
+        return newReview;
+      } else {
+        throw new NotFoundException(
+          'The course id is invalid or the course no longer exists',
+        );
+      }
+    } catch (e) {
+      throw new InternalServerErrorException(e);
+    }
+  }
+
+  // update a Review by Id
+  async updateReview(
+    courseId: Schema.Types.ObjectId,
+    reviewId: Schema.Types.ObjectId,
+    updateReviewDto: UpdateReviewDto,
+  ): Promise<any> {
+    try {
+      const course = await this.CourseModel.findById(courseId);
+      if (course) {
+        let updatedReview = null;
+        updatedReview = await this.ReviewModel.findByIdAndUpdate(
+          reviewId,
+          updateReviewDto,
+          { new: true },
+        );
+        return updatedReview;
```

Comment:
why you have created an extra variable `updatedReview`?

Reply:
> Use `lean` where there is no more operation done on the retrieved Model.
> Learn about lean https://mongoosejs.com/docs/tutorials/lean.html

To return the updated review (if needed)

Reply:
> why you have created an extra variable `updatedReview`?

To return the updated review (if needed)

Reply:
you can return it without assigning it to something. That's only an extra piece of code.

Reply:
Yeah , I'll remove

Reply:
Done